sdk/src/sysvar/recent_blockhashes.rs

```diff
@@ -0,0 +1,89 @@
+use crate::{account::Account, account_info::AccountInfo, hash::Hash, sysvar};
+use bincode::serialized_size;
+use std::ops::Deref;
+
+pub const MAX_ENTRIES: usize = 32;
+const ID: [u8; 32] = [
+    0x06, 0xa7, 0xd5, 0x17, 0x19, 0x2c, 0x56, 0x8e, 0xe0, 0x8a, 0x84, 0x5f, 0x73, 0xd2, 0x97, 0x88,
+    0xcf, 0x03, 0x5c, 0x31, 0x45, 0xb2, 0x1a, 0xb3, 0x44, 0xd8, 0x06, 0x2e, 0xa9, 0x40, 0x00, 0x00,
+];
+
+crate::solana_sysvar_id!(ID, "SysvarRecentB1ockHashes11111111111111111111");
+
+#[repr(C)]
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct RecentBlockhashes(Vec<Hash>);
+
+impl Default for RecentBlockhashes {
+    fn default() -> Self {
+        Self(Vec::with_capacity(MAX_ENTRIES))
+    }
+}
+
+impl RecentBlockhashes {
+    pub fn from_account(account: &Account) -> Option<Self> {
+        account.deserialize_data().ok()
+    }
+    pub fn to_account(&self, account: &mut Account) -> Option<()> {
+        account.serialize_data(self).unwrap();
+        Some(())
+    }
+    pub fn from_account_info(account: &AccountInfo) -> Option<Self> {
+        account.deserialize_data().ok()
+    }
+    pub fn to_account_info(&self, account: &mut AccountInfo) -> Option<()> {
+        account.serialize_data(self).ok()
+    }
+    pub fn size_of() -> usize {
+        serialized_size(&RecentBlockhashes(vec![Hash::default(); MAX_ENTRIES])).unwrap() as usize
+    }
+}
+
+impl Deref for RecentBlockhashes {
+    type Target = Vec<Hash>;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+pub fn create_account(lamports: u64) -> Account {
+    Account::new(lamports, RecentBlockhashes::size_of(), &sysvar::id())
+}
+
+pub fn update_account(account: &mut Account, recent_blockhashes: Vec<Hash>) -> Option<()> {
+    let recent_blockhashes = RecentBlockhashes(recent_blockhashes);
```

Comment:
@rob-solana, ideally, this would be a zero-copy `from_account(account)` and then replace the contents of `RecentBlockhashes.0` with that from the `recent_blockhashes` input. And if so, then there's be no need for that `to_account()` afterward.

Reply:
I wanna see a helper library for zero copy serde

pushing a brand new recent_blockhashes into the accounts_db is faster than reading, updating, storing an update....

Reply:
> pushing a brand new recent_blockhashes into the accounts_db is faster than reading, updating, storing an update....

Unexpected... So https://github.com/solana-labs/solana/pull/6663/commits/486f01790c271cf09ac3dbffba0e40e526e81324 is counterproductive?

Reply:
Probably. I think we need prioritize clear, concise tests and a micro-benchmark, as all this code will change once somebody figures our a good way to do zero-copy deserialization.

Reply:
I went ahead and reverted the fetch+update.  It costs ~1.3us